### DISCUSSION TITLE
Extracting JSON values from my CH column

### DISCUSSION THREAD

**Discussion by Shreya5PGit** [ClickHouse/ClickHouse#55953] [Q&A] [answered]
So im trying to retrieve RMSResponseCode from my JSON formatted column in CH table.
Query is-
SELECT JSONExtractInt(ResponseBody, 'body.RMSResponseCode') AS RMSResponseCode
FROM TradeStationAPIMLog
WHERE (APIName = '/V3/OrderRequest (clone)') AND (ResponseStatus = '200')
LIMIT 30;

This is returning 0 for all entries 
for instance - 
for my corresponding ResponseBody 
{"body":{"AddReqMgn":216202.06000000003,"AvlbMgn":10340.439999999973,"BrokerOrderID":132594322,"ClientCode":"51619763","Exch":"N","ExchOrderID":"","ExchType":"D","LocalOrderID":4,"Message":"Order rejected by RMS as margin required is Rs226542.5, while available margin is Rs10340.44","RMSResponseCode":-7,"ScripCode":52462,"Status":1,"Time":"\/Date(1697481000000+0530)\/"},"head":{"responseCode":"5POrdReqV3","status":"0","statusDescription":"Success"}}

It should have returned -7 as RMSResponseCode column value but it returned 0.

Kindly help me in troubleshooting this

**Comment by den-crane** [comment, marked as answer]
should be `JSONExtractInt(ResponseBody, 'body', 'RMSResponseCode')`

see https://fiddle.clickhouse.com/0d818c3d-3987-4d25-961a-f14130386ca3

**Reply by Shreya5PGit** [reply]
Thanks you so much !
